src-test/src/com/etendoerp/etendorx/datasource/ManageEntityFieldsDSProjectionEntityTest.java

```diff
@@ -0,0 +1,171 @@
+    when(relatedEntity1.getIdentifier()).thenReturn(TestUtils.RELATED_ENTITY_1);
+
+    when(relatedEntity2.getId()).thenReturn("related-2");
+    when(relatedEntity2.getIdentifier()).thenReturn(TestUtils.RELATED_ENTITY_2);
+
+    List<ETRXProjectionEntity> mockRelatedEntities = Arrays.asList(relatedEntity1, relatedEntity2);
+
+    try (MockedStatic<OBContext> obContextMock = mockStatic(OBContext.class)) {
+      try (MockedStatic<OBDal> obDalMock = mockStatic(OBDal.class)) {
+        obDalMock.when(OBDal::getInstance).thenReturn(obDal);
+        when(obDal.getSession()).thenReturn(session);
+        when(session.createQuery(anyString(), eq(ETRXProjectionEntity.class))).thenReturn(query);
+        when(query.setParameter(eq("etrxProjectionEntityId"), any())).thenReturn(query);
+        when(query.list()).thenReturn(mockRelatedEntities);
+
+        @SuppressWarnings("unchecked")
+        List<Map<String, Object>> result = (List<Map<String, Object>>) getProjectionEntityRelatedFilterDataMethod.invoke(
+            manageEntityFieldsDS,
+            mainEntity
+        );
+
+        assertEquals(2, result.size());
+
+        Map<String, Object> firstResult = result.get(0);
+        assertEquals("related-1", firstResult.get("id"));
+        assertEquals(TestUtils.RELATED_ENTITY_1, firstResult.get(TestUtils.NAME));
+        assertEquals(TestUtils.RELATED_ENTITY_1, firstResult.get("_identifier"));
+        assertEquals(ETRXProjectionEntity.ENTITY_NAME, firstResult.get("_entityName"));
+
+        Map<String, Object> secondResult = result.get(1);
+        assertEquals("related-2", secondResult.get("id"));
+        assertEquals(TestUtils.RELATED_ENTITY_2, secondResult.get(TestUtils.NAME));
+        assertEquals(TestUtils.RELATED_ENTITY_2, secondResult.get("_identifier"));
+        assertEquals(ETRXProjectionEntity.ENTITY_NAME, secondResult.get("_entityName"));
+
+        obContextMock.verify(OBContext::setAdminMode);
+        obContextMock.verify(OBContext::restorePreviousMode);
+      }
+    }
+  }
+
+  /**
+   * Tests the retrieval of projection entity related filter data when no results are found.
+   * Verifies that:
+   * - An empty list is returned when no related entities exist
+   * - Admin mode is set and restored properly
+   *
+   * @throws Exception if method invocation fails
+   */
+  @Test
+  public void testGetProjectionEntityRelatedFilterDataNoResults() throws Exception {
+    ETRXProjectionEntity mainEntity = new ETRXProjectionEntity();
+    mainEntity.setId("main-entity-id");
+
+    try (MockedStatic<OBContext> obContextMock = mockStatic(OBContext.class)) {
+      try (MockedStatic<OBDal> obDalMock = mockStatic(OBDal.class)) {
+        obDalMock.when(OBDal::getInstance).thenReturn(obDal);
+        when(obDal.getSession()).thenReturn(session);
+        when(session.createQuery(anyString(), eq(ETRXProjectionEntity.class))).thenReturn(query);
+        when(query.setParameter(eq("etrxProjectionEntityId"), any())).thenReturn(query);
```

Comment:
`etrxProjectionEntityId `
I recommend passing to utils as they are used several times